### PR TITLE
Update Receive page and modal for desktop and mobile

### LINKED ISSUE #217: update receive page & receive modal layout
Use figma!

### LINKED ISSUE #290: [Mobile]: layout receive page
use figma!

### LINKED ISSUE #292: [Mobile]: layout receive modal
use figma!

### LINKED ISSUE #303: Receive page copied button misalignment
image

### PR DESCRIPTION
**Pull request by JustinBeBoy**
Resolve #290 
Resolve  #217 
Resolve #292 
Resolve #303 

**Screenshot**

image

image

### CODE REVIEW

File: ui/cryptomaterial/modal.go [hunk 1]
```diff
@@ -129,6 +131,10 @@ func (m *Modal) Layout(gtx layout.Context, widgets []layout.Widget, width ...flo
 				Top:    unit.Dp(30),
 				Bottom: unit.Dp(30),
 			}
+			insetUniform := layout.UniformInset(m.padding)
```
Comment (ukane-philemon) [inline ui/cryptomaterial/modal.go:134]: `uniformInset`

File: ui/page/root/receive_page.go [hunk 2]
```diff
@@ -168,170 +196,152 @@ func (pg *ReceivePage) generateQRForAddress() {
 	pg.qrImage = &imgdec
 }
 
+func (pg *ReceivePage) getLogo() *cryptomaterial.Image {
```
Comment (ukane-philemon) [inline ui/page/root/receive_page.go:199]: Consider renaming: Something like `selectedWalletLogo` or `getSelectedWalletLogo`?

File: ui/page/root/receive_page.go [hunk 3]
```diff
@@ -168,170 +196,152 @@ func (pg *ReceivePage) generateQRForAddress() {
...
+	if pg.modalLayout != nil {
+		var modalWidth float32 = 450
+		if pg.IsMobileView() {
+			modalWidth = 0
+		}
+		modalContent := []layout.Widget{pg.contentLayout}
+		return pg.modalLayout.Layout(gtx, modalContent, modalWidth)
 	}
-	return pg.layoutDesktop(gtx)
+	return pg.contentLayout(gtx)
```
Comment (ukane-philemon) [inline ui/page/root/receive_page.go:225]: nit: Consider inverting this if statement. It helps with readability to have smaller blocks in if statements.

**Review by ukane-philemon — APPROVED** [review]
Testing well.

File: ui/page/root/receive_page.go [hunk 4]
```diff
@@ -168,170 +196,152 @@ func (pg *ReceivePage) generateQRForAddress() {
...
-	pg.handleCopyEvent(gtx)
-	pg.pageBackdropLayout(gtx)
-
-	if pg.Load.IsMobileView() {
-		return pg.layoutMobile(gtx)
+	if pg.modalLayout == nil {
+		return pg.contentLayout(gtx)
 	}
-	return pg.layoutDesktop(gtx)
+	var modalWidth float32 = 450
```
Comment (ukane-philemon) [inline ui/page/root/receive_page.go:220]: Is this supposed to be a `unit.DP`? Can use the `values.MarginPadding450` var.

Comment (JustinBeBoy) [reply ui/page/root/receive_page.go:220]: It seems that using float32 constants is not really okay but we should fix it later, it is not within the scope of this PR

**Review by dreacot — APPROVED** [review]

**Pull request merged**